### PR TITLE
Deploy on server and check deployment refactoring

### PR DESCRIPTION
**Pull request by jkopriva**
@rhopp please comment:)

### CODE REVIEW

File: plugins/org.jboss.ide.eclipse.as.reddeer/src/org/jboss/ide/eclipse/as/reddeer/server/deploy/DeployOnServer.java [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+		try {
+			// deploy
+			deployProject(projectToDeploy, serverName);
+			// check deploy status
+			checkDeployedProject(projectToDeploy, serverName);
+			// undeploy
+			unDeployModule(projectToDeploy, serverName);
+		} catch (CoreLayerException ex) {
+			new DefaultShell("Server Error");
+			new OkButton().click();
```
Comment (psrna) [inline plugins/org.jboss.ide.eclipse.as.reddeer/src/org/jboss/ide/eclipse/as/reddeer/server/deploy/DeployOnServer.java]: Can you please explain this? It seems you are OK with server error.

Comment (jkopriva) [inline plugins/org.jboss.ide.eclipse.as.reddeer/src/org/jboss/ide/eclipse/as/reddeer/server/deploy/DeployOnServer.java]: Now it should be ok. I have added fail() in catch block.

**Comment by jkopriva** [comment]
test this

**Comment by jkopriva** [comment]
test this

**Comment by jkopriva** [comment]
testPR

**Comment by rhopp** [comment]
testPR

**Comment by rhopp** [comment]
test this

**Comment by jrichter1** [comment]
test this

**Comment by jkopriva** [comment]
test this

**Pull request merged**